File: 167-two-sum-ii-input-array-is-sorted/solution2.cpp
// Runtime: 11ms, Beats 50.63% of users with C++
// Memory: 18.08MB, Beats 9.38% of users with C++

class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        vector<int> res(2);
        int p=0,q=numbers.size()-1;
        while(p<q)
        {
            if(numbers[p]+numbers[q]>target) q--;
            if(numbers[p]+numbers[q]<target) p++;
            if(numbers[p]+numbers[q]==target) break;
        }
        res[0]=p+1;
        res[1]=q+1;
        return res;
    }
};

File: 2864-maximum-odd-binary-number/solution2.py
# Runtime: 34ms, Beats 81.61% of users with Python3
# Memory: 16.58MB, Beats 62.66% of users with Python3

class Solution:
    def maximumOddBinaryNumber(self, s: str) -> str:
        count_one = s.count('1')
        return (count_one-1)*'1'+(len(s)-count_one)*'0'+'1'